lib/bootloader_pvm.pm

```diff
@@ -157,13 +157,20 @@ sub boot_hmc_pvm {
     assert_screen 'pvm-vterm-closed';
 
     # power off the machine if it's still running - and don't give it a 2nd chance
-    type_string "chsysstate -r lpar -m $hmc_machine_name -o shutdown --immed --id $lpar_id && echo 'LPAR SUCCESSFULLY SHUT DOWN'\n";
-    assert_screen [qw(pvm-poweroff-successful pvm-poweroff-not-running)], 180;
-
+    # sometimes lpar shutdown takes long time, we need to check it's state
+    type_string "chsysstate -r lpar -m $hmc_machine_name -o shutdown --immed --id $lpar_id && echo 'LPAR SHUTDOWN NOW'\n";
+    script_run("lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state");
+    assert_screen [qw(lpar-is-running lpar-is-down)];
+    if (match_has_tag('lpar-is-running')) {
+        record_info('LPAR is still running, waiting for shutdown');
+        script_run("for ((i=0\; i<10\; i++)) \; do lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state | grep -q \"Not Activated\" && echo \"LPAR IS DOWN\" && break || echo \"Waiting for lpar $lpar_id to shutdown\" && sleep 5 \; done");
+        save_screenshot;
+    }
     # proceed with normal boot if is system already installed, use sms boot for installation
     my $bootmode = get_var('BOOT_HDD_IMAGE') ? "norm" : "sms";
-    type_string "chsysstate -r lpar -m $hmc_machine_name -o on -b ${bootmode} --id $lpar_id && echo 'LPAR SUCCESSFULLY BOOTED'\n";
-    assert_screen "pvm-poweron-successful";
+    type_string "chsysstate -r lpar -m $hmc_machine_name -o on -b ${bootmode} --id $lpar_id && echo 'LPAR BOOTS UP NOW'\n";
+    script_run("lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state");
+    assert_screen 'lpar-is-running';
```

Comment:
This should be:

- shut down lpar (LPAR SHUTDOWN NOW message can be removed)
- [Wait until lpar is down](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10364#discussion_r431225997)
- check the screen (`assert_screen`)  for the message `LPAR IS DOWN`
- boot lpar (message `LPAR BOOTS UP NOW` can also be removed)
- Wait until lpar is up](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10364#discussion_r431225997) by copy & pasting the for loop, but looking for `Running` instead of  `Not Activated` (see [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10364#pullrequestreview-419312671) and changing the message when the lpar is running
- check the screen (`assert_screen`)  for the message that shows that the lpar is up

Otherwise, you are only checking if the lpar is shut down, instead of waiting for it to be shut down, which results in the same problem: Somebody will have to restart the job, because the lpars were not shut down in the first place

Reply:
updated, thanks